### PR TITLE
Add script to install Elastio packages

### PR DESCRIPTION
**Pull request by e-kov**
This script installs Elastio CLI and/or Elastio-Snap and removes all installed Elastio packages.

One-liner for installation of the Elastio CLI and Elastio-Snap:
```bash
sudo /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh)"
```
One-liner for uninstallation:
```bash
curl -Os https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh && chmod +x install-elastio.sh && sudo ./install-elastio.sh -u
```

One-liner for Elastio CLI installation only:
```bash
curl -Os https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh && chmod +x install-elastio.sh && sudo ./install-elastio.sh -c
```
One-liner for Elastio-Snap installation only:
```bash
curl -Os https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh && chmod +x install-elastio.sh && sudo ./install-elastio.sh -d
```

This script [install_driver.sh](https://github.com/elastio/elastio/blob/master/yozhik/tests/linux/install_driver.sh) and this [instruction](https://github.com/elastio/packaging/blob/master/INSTALL.md) were taken as basis.

*NOTE:*
The links above will be changed from the  `...feat/1418-install-elastio-packages...` to the `...master...` after merge of the PR to `master`.

Resolves elastio/elastio#1418

### CODE REVIEW

**Comment by anelson** [comment]
Can you describe all of those examples in a single command rather than chaining the script download, `chmod`, and invocation?  Whenever we do that it seems to confuse Naj.

File: scripts/install-elastio.sh [hunk 1]
```diff
@@ -0,0 +1,229 @@
+#!/bin/bash
+
+me=$(basename $0)
+dir=$(readlink -f $(dirname $0))
+
+kernel_devel_install()
```
Comment (anelson) [inline scripts/install-elastio.sh:6]: This function name sounds generic, but it uses `yum` so it's CentOS/RHEL specific.  Don't you think that will be confusing to a future reader?

File: scripts/install-elastio.sh [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+{
+    echo "This script installs latest Elastio CLI and Elastio-Snap packages if executed without arguments, or re-installs them, if they are already installed."
+    echo "All commandline arguments are optional."
+    echo "Usage examples:"
+    echo "   $me"
+    echo "   $me -u"
+    echo "   $me --cli-only"
+    echo "   $me --driver-only"
+    echo
+    echo "  -c | --cli-only       : Install Elastio CLI only, without changeds-tracking driver Elastio-Snap."
```
Comment (anelson) [inline scripts/install-elastio.sh:102]:
```suggestion
    echo "  -c | --cli-only       : Install Elastio CLI only, without change tracking driver elastio-snap."
```

File: scripts/install-elastio.sh [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    echo
+    echo "  -d | --driver-only    : Install changeds-tracking driver Elastio-Snap only, without Elastio CLI."
```
Comment (anelson) [inline scripts/install-elastio.sh:104]:
```suggestion
    echo "  -d | --driver-only    : Install change tracking driver elastio-snap only, without Elastio CLI."
```

File: scripts/install-elastio.sh [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    echo "                          NOTE: There is no Elastio-Snap packages for Fedora (kernel 5.8 and newer) for now."
+    echo "                                So, temporarily this option does nothing on Fedora."
```
Comment (anelson) [inline scripts/install-elastio.sh:106]:
```suggestion
    echo "                          NOTE: There are no elastio-snap packages for Fedora (kernel 5.8 and newer) as of yet."
    echo "                                So, temporarily this option does nothing on Fedora."
```

File: scripts/install-elastio.sh [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+            exit 1
+        fi
+        cent7_amazon_install Amazon $(rpm -E %amzn) amzn
+    ;;
+
+    scientific | sl | oracle | ol )
+        echo "Warning: Oracle Linix and Scientific Linux are not officially supported!"
+        if [ -z $force ]; then
+            echo "We can try to install packages for CentOS on your system."
+            echo "Add '--force' to insist on the installation. But keep in mind, you act at your own risk!"
```
Comment (anelson) [inline scripts/install-elastio.sh:170]:
```suggestion
            echo "Add '--force' to insist on the installation. But beware this isn't officially supported!"
```

File: scripts/install-elastio.sh [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+            exit 1
+        fi
+    ;;&
+
+    centos | el | rhel | red | scientific | sl | oracle | ol )
+        case ${dist_ver} in
+            7 | 8 ) cent_fedora_install CentOS $(rpm -E %rhel) el ;;
+            * )
+                echo "CentOS/RHEL and derivatives like Oracle Linux and Scientific Linux 7 and 8 are supported. Current distro version $dist_ver isn't supported."
```
Comment (anelson) [inline scripts/install-elastio.sh:179]: You just said above that Oracle and SL are not officially supported.

```suggestion
                echo "Old versions of CentOS/RHEL and derivatives like Oracle Linux and Scientific Linux 7 and 8 are not supported. This includes your current distro version $dist_ver."
```

File: scripts/install-elastio.sh [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+                exit 1
+            ;;
+        esac
+    ;;
+
+    fedora | fc )
+        if [ ! -z "$driver" ]; then
+            echo "There is no Elastio-Snap packages for Fedora (kernel 5.8 and newer) for now. Ignoring driver installation..."
```
Comment (anelson) [inline scripts/install-elastio.sh:187]:
```suggestion
            echo "There are no elastio-snap packages for Fedora (kernel 5.8 and newer) yet. Ignoring driver installation..."
```

**Review by anelson — CHANGES_REQUESTED** [review]

File: scripts/install-elastio.sh (hunk 6 above)
Comment (e-kov) [reply scripts/install-elastio.sh:179]: This message is about versions out of supported range like CentOS 5, 6, 9 ))
It'll be shown for them. The suggested phrase is confusing, especially "7 and 8 are *not* supported"

Comment (e-kov) [reply scripts/install-elastio.sh:179]: Let's say shortly just "CentOS/RHEL versions 7 and 8 are supported. Current distro version $dist_ver isn't supported."

File: scripts/install-elastio.sh (hunk 1 above)
Comment (e-kov) [reply scripts/install-elastio.sh:6]: Fedora too

Comment (e-kov) [reply scripts/install-elastio.sh:6]: renamed kernel_devel_install -> cent_fedora_kernel_devel_install

**Comment by e-kov** [comment]
>Can you describe all of those examples in a single command rather than chaining the script download, chmod, and invocation? Whenever we do that it seems to confuse Naj.

There is a problem with the `x` attribute and download from the `raw.githubusercontent.com`. The downloaded script doesn't have this attribute.

So truthful one-liner with the single command is only for the script execution w/o arguments like this:
```bash
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/.../script.sh)"
```
where bash takes some text as executable argument. And this text is the content of the script produced by `curl`. There is a pretty ugly way to add arguments. Something like 
```bash
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/.../script.sh $0)" -- --script-argument
```

Thus for the options with the arguments, I had to download the script, fix `x` attribute by `chmod` and then execute with an argument (in 3 hops) assuming it's less confusing:
```bash
curl -Os https://raw.githubusercontent.com/.../script.sh && chmod +x script.sh && sudo ./script.sh --finally-an-argument
```
----
I just found another way to workaround arguments without saving script.As an example, for the `--driver-only` argument it lookslike this:
```bash
curl -fsSL https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh | sudo bash /dev/stdin --driver-only
```

---
So I have 3 options for you:
1) as it was proposed in description (3 hops for args):
```bash
# Install Elastio CLI and Elastio-Snap:
sudo /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh)"

# Uninstall all Elastio packages:
curl -Os https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh && chmod +x install-elastio.sh && sudo ./install-elastio.sh -u

# Install Elastio CLI only:
curl -Os https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh && chmod +x install-elastio.sh && sudo ./install-elastio.sh -c

# Install Elastio-Snap only:
curl -Os https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh && chmod +x install-elastio.sh && sudo ./install-elastio.sh -d
```
2) `/bin/bash -c` and args:
```bash
# Install Elastio CLI and Elastio-Snap:
sudo /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh)"

# Uninstall all Elastio packages:
sudo /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh) $0" -- -u

# Install Elastio CLI only:
sudo /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh) $0" -- -c

# Install Elastio-Snap only:
sudo /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh) $0" -- -d
```
3) `curl` and `bash` in pipe + `/dev/stdin`:
```bash
# Install Elastio CLI and Elastio-Snap:
curl -fsSL https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh | sudo bash

# Uninstall all Elastio packages:
curl -fsSL https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh | sudo bash /dev/stdin -u

# Install Elastio CLI only:
curl -fsSL https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh | sudo bash /dev/stdin -c

# Install Elastio-Snap only:
curl -fsSL https://raw.githubusercontent.com/elastio/elastio-stack/feat/1418-install-elastio-packages/scripts/install-elastio.sh | sudo bash /dev/stdin -d
```

Please chose one which is less confusing in your opinion.

File: scripts/install-elastio.sh [hunk 8]
```diff
@@ -0,0 +1,228 @@
+#!/bin/bash
```
Comment (Veetaha) [inline scripts/install-elastio.sh:1]: Ugh, I am looking forward to having `elastioup` Rust binary in this repo instead :sweat:

File: scripts/install-elastio.sh (hunk 6 above)
Comment (e-kov) [reply scripts/install-elastio.sh:178]: Using this phrase and re-requesting review.

File: scripts/install-elastio.sh (hunk 8 above)
Comment (e-kov) [reply scripts/install-elastio.sh:1]: ...and a bash script to install/uninstall that rust binary? ))

Comment (Veetaha) [reply scripts/install-elastio.sh:1]: @e-kov 
We can just give direct links to compiled binaries for each platform as terraform does: https://www.terraform.io/downloads.html

Comment (e-kov) [reply scripts/install-elastio.sh:1]: here we only have 1 installation link for all supported distributions. and 3 arguments for uninstall and 2 rare cases. not much to say that it is difficult. don't recall why you say "ugh" )

Comment (Veetaha) [reply scripts/install-elastio.sh:1]: I say "ugh" just when I see bash in general.
Do we support installation on windows?

Comment (anelson) [reply scripts/install-elastio.sh:1]: No, Windows doesn't even have the same shell, so the Windows script will always be different.

**Review by anelson — APPROVED** [review]

**Pull request merged**